devito/function.py

```diff
@@ -229,7 +247,7 @@ def shape_data(self):
         """
         Full allocated shape of the data associated with this :class:`Function`.
         """
-        return self.shape_domain
+        return tuple(i - s for i, s in zip(self.shape_domain, self.staggered))
```

Comment:
so indices are `i-1` not 1/2?

Reply:
No, this has nothing to do with indices; these are data shapes. In every dimension you discretize on a facet/edge/cell, you need one less data point.

Reply:
My bad, hidden line, saw this with `_indices`. So I guess this PR is shape only, not staggered FD yet?

Reply:
Correct. The symbolics will get their own (set of) PR(s).